Doc generation and benchmark modules (#4151)
SpikeInterface/spikeinterface: doc/references.rst

If you use :code:`auto_label_units` or :code:`train_model`, please cite [Jain]_

Benchmark
---------

> For adicts to benchmarking numerical methods in the context of spike sorting, if any attends to use the benchmark

Reviewer: ```suggestion
For addicts to benchmarking numerical methods in the context of spike sorting, if any intends to use the benchmark
```

> For addicts to benchmarking numerical methods in the context of spike sorting, if any intends to use the benchmark

Reviewer: ```suggestion
For addicts to benchmarking numerical methods in the context of spike sorting, if you use the benchmark
```
